### PR TITLE
kv/client: don't resolve lock immediately after a region is initialized

### LINKED ISSUE #2188: kv client resolves lock immediately after ticdc is up
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

https://github.com/pingcap/ticdc/blob/b6451b6481538a92ddfdfe6de5a4901789e0b159/cdc/kv/region_worker.go#L293-L315

```
$ less cdc.log
[2021/06/29 15:40:31.453 +08:00] [WARN] [region_worker.go:303] 
["region not receiving resolved event from tikv or resolved ts is not pushing for too long time, try to resolve lock"] 
[regionID=1031073] 
[span="[7480000000000023ffe95f728000000000ff0021960000000000fa, 7480000000000023ffe95f728000000000ff00b70e0000000000fa)"] 
[duration=74h26m11.05s] [lastEvent=821.902081ms] [resolvedTs=425901283045015552]

$ cdc cli changefeed list
  {
    "id": "bl1",
    "summary": {
      "state": "normal",
      "tso": 425901283045015552,
      "checkpoint": "2021-06-26 13:14:19.858",
      "error": {
        "addr": "172.16.5.33:47912",
        "code": "CDC:ErrOwnerUnknown",
        "message": "rpc error: code = DeadlineExceeded desc = context deadline exceeded"
      }
    }
  }
```

2. What did you expect to see?

Do not resolve lock so quickly.

3. What did you see instead?

4. Versions of the cluster

Release-5.0 branch b6451b6481538a92ddfdfe6de5a4901789e0b159

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Fix #2188

### What is changed and how it works?

Introduce a penalty mechanism
- If TiCDC keeps receiving unchanged resolved-ts, increase the penalty. 
- If TiCDC receives a new resolved ts, reset the penalty to zero
- Only resolve ts if the penalty is larger than threshold

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- Don't resolve lock immediately after a region is initialized.

### CODE REVIEW

File: cdc/kv/region_worker.go [hunk 1]
```diff
@@ -304,6 +307,17 @@ func (w *regionWorker) resolveLock(ctx context.Context) error {
 							zap.Duration("duration", sinceLastResolvedTs), zap.Duration("since last event", sinceLastResolvedTs))
 						return errReconnect
 					}
+					// Only resolve lock if the resovled-ts keeps unchanged for
+					// more than resolveLockPenalty times.
+					if rts.ts.penalty < resolveLockPenalty {
+						if lastResolvedTs > rts.ts.resolvedTs {
+							rts.ts.resolvedTs = lastResolvedTs
+							rts.ts.eventTime = time.Now()
+							rts.ts.penalty = 0
```
Comment (overvenus) [inline cdc/kv/region_worker.go:316]: Could you reset the penalty in `func Upsert()`?

**Review by overvenus — APPROVED** [review]

File: cdc/kv/region_worker.go (hunk 1 above)
Comment (amyangfei) [reply cdc/kv/region_worker.go:316]: `Upsert` here is a `insert` operation actually, so we must reset penalty outside

**Review by liuzix — APPROVED** [review]

**Pull request merged**